src/main/app/src/app/zaken/zaak-create/zaak-create.component.ts

```diff
@@ -243,24 +241,36 @@ export class ZaakCreateComponent implements OnInit, OnDestroy {
     if (formGroup) {
       const zaak: Zaak = new Zaak();
       Object.keys(formGroup.controls).forEach((key) => {
-        if (key === "vertrouwelijkheidaanduiding") {
-          zaak[key] = formGroup.controls[key].value?.value;
-        } else if (key === "initiatorIdentificatie" && this.initiator != null) {
-          zaak["initiatorIdentificatieType"] = this.initiator.identificatieType;
-          zaak[key] = this.initiator.identificatie;
-        } else if (key === "bagObjecten") {
-          // skip
-        } else if (key === "toekenning") {
-          if (this.medewerkerGroepFormField.formControl.value.medewerker) {
-            zaak.behandelaar =
-              this.medewerkerGroepFormField.formControl.value.medewerker;
+        Object.keys(formGroup.controls).forEach((key) => {
+          switch (key) {
+            case "vertrouwelijkheidaanduiding":
```

Comment:
I am not sure about these static case values, I'd much rather use a "code", or as a minimum pull these values from a properties (translation) file.

Reply:
Agreed; but my only aim was readability of existing code. It is a bit of a tricky thing here. Looks like the aim was making a list of fields and just submit. But then there needed to be some enhancement etc on certain fields.

Making this properly (or better coded) here should be used an enum object including callback functions performing the special handling. Great improvement!! But I prefer to do that in separate task/story. OK by @bas-info-nl ? 😉

Reply:
BTW; Same code setup for going over the form controls in two other files:

src/main/app/src/app/informatie-objecten/informatie-object-add/informatie-object-add.component.ts
src/main/app/src/app/informatie-objecten/informatie-object-edit/informatie-object-edit.component.ts

Reply:
Story Created: https://dimpact.atlassian.net/browse/PZ-3722